doubly_linkedlist.py

```diff
@@ -0,0 +1,86 @@
+class Dll:
+    def __init__(self, values=None):
+        self.head = None
+        self._length = 0
+        if values:
+            for value in values:
+                self.head = Node(value, self.head, None)
+                if self.head.next:
+                    self.head.next.previous = self.head
+                self._length += 1
+
+    def push(self, val):
+        self.head.next = self.head
+        self.head = Node(val, self.head, None)
+        self._length += 1
```

Comment:
To me this doesn't look like it's adjusting the .previous's correctly, and your tests are insufficient to convince me.